src/kernel/adam_step/cuda.cu

```diff
@@ -20,22 +20,30 @@ namespace nntile::kernel::adam_step
 
 template<typename T>
 static __global__
-void cuda_kernel(Index num_iter, Index num_elems, T beta_1, T beta_2, T eps,
-        T lr, T weight_decay, const T *grad, T *first_moment, T *second_moment,
-        T *p, T alpha, T beta)
+void cuda_kernel(Index num_iter, Index num_elems, Scalar beta_1_, Scalar beta_2_, Scalar eps_,
+        Scalar lr_, Scalar weight_decay_, const T *grad, T *first_moment, T *second_moment,
+        T *p)
 {
     int i = threadIdx.x + blockIdx.x*blockDim.x;
+    using Y = typename T::repr_t;
+    const Y beta_1{beta_1_};
+    const Y beta_2{beta_2_};
+    const Y lr{lr_};
+    const Y eps{eps_};
+    const Y weight_decay{weight_decay_};
+    const Y alpha = lr / (Y{1.0} - std::pow(beta_1, num_iter));
```

Comment:
remove std namespace. These functions (pow, sqrt) must come from NVCC compiler and not from math.h